alpaca/rest.go

```diff
@@ -30,6 +30,8 @@ type ClientOpts struct {
 	RetryDelay   time.Duration
 	// HTTPClient to be used for each http request.
 	HTTPClient *http.Client
+	// ResponseCallback is a user defined callback to execute on the raw http.Response
+	ResponseCallback func(*http.Response)
```

Comment:
It would be useful to also provide the request.

```suggestion
	ResponseCallback func(*http.Request, *http.Response)
```

Reply:
Indeed, I agree it can also be useful, I'll wait for the maintainers feedback and implement this accordingly.

Reply:
@gcatlin You're the original reporter of https://github.com/alpacahq/alpaca-trade-api-go/issues/292. If this solution works for you, I'm fine with it with one minor suggestion: if both the request and the response is in the callback then could you maybe call it `HTTPCallback` instead of `ResponseCallback`?

Reply:
Yeah, this solution works for me. `HTTPCallback` works too.

Reply:
Will add the request and rename to HTTPCallback than 👍